File: 883/e.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    map<int, int> arr;
    cout << arr[0] << '\n';
    for (int k = 2; k < 1'000'000; k++) {
        long long n = 1;
        long long cp = 1;
        while (n <= 1'000'001) {
            arr[n] = 1;
            cp*= k;
            n+= cp;
            // cout << n << '\n';
        }
        // cout << k << '\n';
    }

    cout << "done\n";

    // int n; cin >> n;
    // for (int i = 0; i < n; i++) {
    //     int d; cin >> d;
    //     if (arr[d] == 1) cout << "YES\n";
    //     else cout << "NO\n";
    // }
}

File: 883/e.py
arr = [0 for _ in range(int(1e6+10))]

for k in range(2, int(1e6+10)):
    n = 1
    cp = 1
    done = False
    done2 = False
    while n <= 1000001:
        if done2:
            arr[n] = 1
        cp*= k
        n+= cp
        if done:
            done2 = True
        done = True

for _ in range(int(input())):
    if arr[int(input())]: print("YES")
    else: print("NO")